app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -498,6 +501,23 @@ private void updateTheDarkness() {
         image.setAlpha(1.0f - scrollPercentage);
     }
 
+    /**
+    * Returns captions for media details
+     *
+     * @param media object of class media
+     * @return caption as string
+     */
+
+    private String prettyCaption(Media media) {
+        String caption = media.getCaption().trim();
+        if (caption.equals("")) {
+            return getString(R.string.detail_caption_empty);
```

Comment:
If possible, remove the whole UI row instead.

Reply:
okay I'll do in the next  commit

Reply:
Is this solved? Is an empty row shown in some cases?